src/wmtk/invariants/TetrahedronInversionInvariant.cpp

```diff
@@ -0,0 +1,37 @@
+#include "TetrahedronInversionInvariant.hpp"
+#include <wmtk/Mesh.hpp>
+#include <wmtk/simplex/faces_single_dimension.hpp>
+#include "predicates.h"
+
+namespace wmtk {
+TetrahedronInversionInvariant::TetrahedronInversionInvariant(
+    const Mesh& m,
+    const MeshAttributeHandle<double>& coordinate)
+    : Invariant(m)
+    , m_coordinate_handle(coordinate)
+{}
+
+bool TetrahedronInversionInvariant::after(PrimitiveType type, const std ::vector<Tuple>& tets) const
+{
+    if (type != PrimitiveType::Tetrahedron) return true;
```

Comment:
Why not an exception? This should not happen.

Reply:
the generic underlying invariant calls basically invoke `after` on every type of simplex possible. We could/should decrease this by letting invariants register which dimensions they care about.